apps/src/templates/instructions/codeReview/Comment.jsx

```diff
@@ -20,17 +20,15 @@ class Comment extends Component {
     viewAsTeacher: PropTypes.bool
   };
 
-  state = {isShowingCommentOptions: false};
-
-  onDelete = () => {
-    this.setState({isShowingCommentOptions: false});
-    this.props.onDelete();
+  state = {
+    hideWhileResolved: true
   };
 
-  onResolve = () => {
-    this.setState({isShowingCommentOptions: false});
-    this.props.onResolveStateToggle();
-  };
+  componentDidUpdate(prevProps) {
+    if (!prevProps.comment.isResolved && this.props.comment.isResolved) {
```

Comment:
why do we only need to check the transition from not resolved to resolved?

Reply:
Setting the comment to be hidden only matters if the comment is resolved. Setting it when the comment become unresolved would be a no-op, but would cause the component to re-render.